docs/proposals/scheduling/20220701-pod-migration-job.md

```diff
@@ -0,0 +1,397 @@
+- Sort by the number of replicas being migrated by the workload. The lower the number of replicas being migrated, the priority will be given to processing.
+
+##### Execute PodMigrationJob
+
+- Update PodMigrationJobStatus.Phase to Running to trigger the PodMigrationJob controller reconcile these jobs
+- PodMigrationJob controller reconciles process:
+  - If the mode of PodMigrationJob is `EvictionDirectly`, just delete the Pod through the delete method that configured in PodMigrationJob controller. And update the phase of PodMigrationJob to Success.
+  - If not specified ReservationOptions.ReservationRef, create the Reservation instance by the reservation template or Pod  spec to reserve resources. And updates the created Reservation instance to the ReservationOptions.ReservationRef.
+  - Check the status of Reservation to determine whether reserve resource successfully.
+  - If failed to reserve, abort the PodMigrationJob and update the phase of PodMigrationJob to Fail
+  - If successfully reserve, delete the Pod through the delete method that configured in PodMigrationJob controller.
+  - Check the Reservation status to determine whether the Reservation consumed.
+  - If Reservation consumed, tracks the status of Reservation and update the status to PodMigrationJob
+  - Update phase of PodMigrationJob to Success.
+
+##### Migration Stability mechanism
+
+- Support for disabling this capability by configuration 
+- Supports a simple central flow control mechanism to limit the number of migrations over a period of time. 
+
+See the Configuration section for more details
+
+#### Controller Configuration
+
+User can configure the `PodMigrationJobControllerConfiguration` through Koordinator Descheduler ConfigMap. 
+
+```go
+type PodMigrationJobControllerConfiguration struct {
+  // Paused indicates whether the PodMigrationJob Controller should to work or not.
+  Paused bool `json:"paused,omitempty"`
+  // DryRun means only execute the entire migration logic except create Reservation or Delete Pod
+  // Default is false
+  DryRun bool `json:"dryRun,omitempty"`
+
+  // FlowControlQPS controls the number of arbitrations per second
+  FlowControlQPS string `json:"flowControlQPS,omitempty"`
+  // FlowControlBurst is the maximum number of tokens
+  FlowControlBurst int32 `json:"flowControlBurst,omitempty"`
+
+  // MaxMigratingPerNode represents he maximum number of pods that can be migrating during migrate per node.
+  MaxMigratingPerNode *int32 `json:"maxMigratingPerNode,omitempty"`
+
+  // MaxMigratingPerNamespace represents he maximum number of pods that can be migrating during migrate per namespace.
+  MaxMigratingPerNamespace *int32 `json:"maxMigratingPerNamespace,omitempty"`
+
+  // MaxMigratingPerWorkload represents he maximum number of pods that can be migrating during migrate per workload.
+  MaxMigratingPerWorkload *intstr.IntOrString `json:"maxMigratingPerWorkload,omitempty"`
```

Comment:
Is this value a ratio or?

Reply:
Want to support usage scenarios like 10% replicas or 5 replicas. I'll update the comment.

Reply:
done.

Reply:
done.